vividus/src/main/java/org/vividus/spring/SubscriberRegisteringBeanPostProcessor.java

```diff
@@ -29,13 +29,16 @@
 {
     @Inject private EventBus eventBus;
 
+    @SuppressWarnings("unchecked")
     @Override
     public Object postProcessAfterInitialization(Object bean, String beanName)
     {
-        if (Stream.of(bean.getClass().getDeclaredMethods()).anyMatch(m -> m.isAnnotationPresent(Subscribe.class)))
-        {
-            eventBus.register(bean);
-        }
+        Stream.iterate((Class<Object>) bean.getClass(), c -> c.getSuperclass() != null, Class::getSuperclass)
+              .map(Class::getDeclaredMethods)
```

Comment:
what about just `getMethods`?

Reply:
sure, I don't know why it was getDeclaredMethods previously, but from docs event bus works with public handler methods